### ISSUE TITLE
Why doesn't Marshmallow Dict field resolve to additionalProperties?

### ISSUE THREAD

**Issue by UrKr** [marshmallow-code/apispec#201]
Any Dict field in a schema seems to resolve to just 'type': 'object' in the spec output. Why not to additionalProperties? 

```

class ASchema(MM.Schema):
    data = MM.fields.Dict(keys=MM.fields.String(), values=MM.fields.Nested(OtherSchema), required=True)
```

the yaml output for:
`
spec.definition('A', ASchema, schema=ASchema)
`
is this: 
`data: {type: object}`
Why not this:
```
data: {
  type: object,
  additionalProperties: {
    $ref: '#/definitions/A'
  }
}
```

**Comment by lafrech** [comment]
`keys` and `values` were added in marshmallow 3 and are not yet documented by apispec. Indeed, it would be nice to have them documented.

Not sure how that would look like, though.

(Thinking out loud.) In marshmallow, keys can be anything, including stuff that would generate wrong JSON. (I mean can you use arrays as dict indexes in JSON? I guess not.) It is fine if we ignore pathological cases. I mean if the doc feature does not work on cases that can't be functional anyway, no big deal.

**Comment by UrKr** [comment]
Yes, I suppose it would only have to work with string keys. Though the `keys` value could actually just be ignored, no? 

Are you taking pull requests for this? I don't have one ready or anything, I would have to look at the code and see what I can do. Is there any other ambiguity apart from the keys? Naively, I would think it's just a matter of adding additional `Dict` field handling.

**Comment by lafrech** [comment]
Yes, I suppose we can ignore `keys`.

We welcome PRs. When the change involved is a bit tricky, it can be better to discuss it here before actually working on it, to avoid the frustration of seeing the PR not merged.

From this [SO post](https://stackoverflow.com/questions/41239913/why-additionalproperties-is-the-way-to-represent-dictionary-map-in-swagger-ope), it seems your fix is correct, so it's simpler than I thought. You can send a PR implementing this.

Thanks!

**Comment by lafrech** [comment]
Fixed in #202.
